android/app/src/main/java/au/com/robin/sms/service/SubscriberService.kt

```diff
@@ -38,6 +39,8 @@ private const val NOTIFICATION_GROUP_ID = "com.robin.sms.NOTIFICATION_GROUP_SERV
  * - https://robertohuertas.com/2019/06/29/android_foreground_services/
  */
 class SubscriberService : Service() {
+    private var repository: Repository? = null
+    private var wsConnection: WsConnection? = null
```

Comment:
I think we can avoid making wsConnection nullable by instantiating wsConnection in `onStartCommand` instead of as a SubscriberService class member.
```kotlin
onStartCommand() {
  val wsConnection = WsConnection(...)
  when (intent.action) {
    Actions.START.name -> starService(wsConnection)
    Actinos.STOP.name -> stopService(wsConnection)
  }
}

```

Reply:
Good catch! Yes, I think it is much better that way. I'll change it

Reply:
I just realised that we also want to stop the service when `onDestroy` is called. 
![image](https://github.com/robinchew/android-sms-server/assets/48295101/3ac817b0-cfa1-48ac-9498-153e5eab337c)
So, maybe just keep WsConnection nullable?

Reply:
ok sure if you cant do anything about it

Reply:
when does subsriber service gets destroyed

Reply:
it gets destroyed when user explicitly force stop the app through the app setting and if android is really low in resources as well as when `stopSelf()` is called. So, in our case, we are highly likely going to explicitly stop our service through our custom `stopService()` which means we can pass in WsConnection's instance as parameter as we want to keep listening to the server unless we want it stop. But, do you think leaving out `onDestroy` is a good idea?

Reply:
Nevermind then, I'll stop fighting android's stupid framework.

Reply:
Looks like im still fighting, check the following code out:
```kt
class Repository(val abc: String = "abc") {
}

open class Service() {}

class SubscriberService(
    val repo: Repository = Repository("xyz") // <-- Do this?
) : Service() {
    fun onCreate() {
    }
}
fun main() {
    val abc = SubscriberService().repo.abc
    println("Hello, World! ${abc}")
}
```

Reply:
Ignore that, not a good example because `applicationContext` needs to be passed to `Repository`.